web-components/src/components/card/Card.ts

```diff
@@ -1,53 +1,153 @@
+      this.dispatchEvent(
+        new CustomEvent<{ id: string; type: string }>("card-click", {
+          detail: {
+            id: this.id,
+            type: ev
+          },
+          bubbles: true,
+          composed: true
+        })
+      );
+    }
+
+    handleCardKeyDownEvent(event: KeyboardEvent, label: string) {
+      this.dispatchEvent(
+        new CustomEvent<{ id: string; type: string }>("card-keydown", {
+          detail: {
+            id: this.id,
+            type: label.toLowerCase()
+          },
+          bubbles: true,
+          composed: true
+        })
+      );
+    }
+
     render() {
       return html`
-        <div class="md-card" part="card">
+        <div class="md-card ${classMap(this.cardClassMap)}" id="${this.id}" part="card">
           <div class="md-card-header">
-            <slot name="header">
-              <md-avatar alt="avatar" label="avatar" title=${this.title} src=${this.src}></md-avatar>
-              <div class="md-card-header-title">
-                <h3>${this.title}</h3>
-                <h5>${this.subtitle}</h5>
-              </div>
-              <div class="md-card-header-actions">
-                <md-menu-overlay placement="bottom-end">
-                  <md-button variant="white" slot="menu-trigger" circle>
-                    <md-icon slot="icon" name="icon-more-adr_16"></md-icon>
-                  </md-button>
-                  <md-list></md-list>
-                </md-menu-overlay>
-              </div>
+            <slot name="card-header-icon">
+              <md-favorite value="${`card-` + this.id}"></md-favorite>
             </slot>
+            <div class="md-card-header-title">
+              <slot name="card-header-title">
+                <h1>${this.title}</h1>
+                <h2>${this.subtitle}</h2>
+              </slot>
+            </div>
+            <div class="md-card-header-actions">
+              ${this.info
+                ? html`
+                    <md-tooltip message="${this.info}" placement="bottom">
+                      <md-button ariaLabel="Card Info" class="md-card-info-icon" color="color-none" size="size-none">
+                        <md-icon slot="icon" name="info_16"></md-icon>
+                      </md-button>
+                    </md-tooltip>
+                  `
+                : nothing}
+              ${this.fullscreen
+                ? html`
+                    <md-button
+                      @button-click=${this.handleToggleExpandCollapse}
+                      ariaLabel="Maximaze Card"
+                      class="md-card-max-icon"
+                      color="color-none"
+                      size="size-none"
+                    >
+                      <md-icon slot="icon" name=${this.full ? "minimize_16" : "maximize_16"}></md-icon>
+                    </md-button>
+                  `
+                : nothing}
+              <md-menu-overlay class="md-card-menu" placement="bottom-start">
```

Comment:
we should only show it if `this.menuOption` is not empty

Reply:
Done